crates/storage-api/src/lib.rs

```diff
@@ -38,12 +38,36 @@ pub mod service_status_table;
 pub mod state_table;
 pub mod timer_table;
 
+/// Level of read isolation of a storage transaction
+pub enum ReadIsolation {
+    /// No read isolation. This allows the storage implementation to return dirty reads. Use this
+    /// level, if you know that you are the only writer.
+    None,
+    /// Reads are served from a consistent snapshot of the underlying storage implementation. This
+    /// ensures that multiple reads get a consistent view of the stored information. Use this level,
+    /// if you need to do multiple reads and tolerate concurrent write operations.
+    ///
+    /// Note that using this level is more costly than [`ReadIsolation::None`].
+    Snapshot,
+}
```

Comment:
I'm not sure if the documentation matches reality here. What do we mean by "dirty reads" here? The default is that the writebatch/tx's changes is not visible to other transactions unless it's committed.

Do we have other uses of the API that split a single logical operation across multiple write batches?

Reply:
This is true for the concrete implementation of our `Storage` trait. I just didn't want to overfit to it (even though there is probably no other impl in the foreseeable future).

Reply:
I suggest we call this `IsolationLevel` and maybe rename `Snapshot` to `RepeatableReads`?

Reply:
Sounds fair. Will update it accordingly.